### ISSUE TITLE
NET-1685 Analysis warnings in .razor files cannot be suppressed

### ISSUE THREAD

**Issue by costin-zaharia-sonarsource** [SonarSource/sonar-dotnet#8050]
### Description

There are multiple ways to suppress analysis warnings:
- using `.editorconfig` files
- using `#pragma warning disable` directives
- using the `SupressMessage` attribute
Details: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/suppress-warnings

Currently none of these work for `.razor` files.

### Repro steps

Create a project with a `.razor` file that contains the following snippet:
```
@code {
#pragma warning disable S5332
    public void Do()
    {
        using var smtp = new SmtpClient("host", 25);
    }
#pragma warning restore S5332
}
```

### Expected behavior

The issue should be suppressed.

### Actual behavior

The impression does not work.

### Known workarounds

Mark the issue as won't fix/false positive or review the hotspot on the server

### Related information

* C#/VB.NET Plugins version 9.10
* MSBuild version 17.7.3+8ec440e68

**Comment by Piedone** [comment]
The same is true for Razor templates in `.cshtml` files as well.

**Comment by jeanbeziaud-bib** [comment]
Another workaround is to disable the Razor code analyze:

![Image](https://github.com/user-attachments/assets/3b780b36-21ce-4f7d-9bc1-58f56415e4b6)

**Comment by jeanbeziaud-bib** [comment]
I confirm: via .editor.config, it doesn't work too with this configuration:

![Image](https://github.com/user-attachments/assets/485ab16d-cce6-4cc5-bb92-ed5da75d53cd)

**Comment by pavel-mikula-sonarsource** [comment]
Moved to [NET-1685](https://sonarsource.atlassian.net/browse/NET-1685)

[NET-1685]: https://sonarsource.atlassian.net/browse/NET-1685?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by Piedone** [comment]
@pavel-mikula-sonarsource is this indeed completed, or did you close it because of the move? Asking because the latest release (https://github.com/SonarSource/sonar-dotnet/releases/tag/10.11.0.117924), published after this being closed, doesn't seem to contain it.

**Comment by pavel-mikula-sonarsource** [comment]
It was only moved. I'll reopen it so you can get an update once it's done. We've kept opened mostly FPs

**Comment by Piedone** [comment]
Thank you!
